infra/src/main/kotlin/fi/fta/geoviite/infra/ratko/RatkoClient.kt

```diff
@@ -392,10 +392,11 @@ class RatkoClient @Autowired constructor(val client: RatkoWebClient) {
                     "secondarySortOrder" to "ASC"
                 )
             ))
-            pagePoints = (ratkoJsonMapper.readValue(body, RatkoOperatingPointAssetsResponse::class.java)?.assets
-                ?: listOf()).mapNotNull(::parseAsset)
-            allPoints.addAll(pagePoints)
-        } while (pagePoints.size == 100)
+            val allAssetsInPage =
+                ratkoJsonMapper.readValue(body, RatkoOperatingPointAssetsResponse::class.java)?.assets ?: listOf()
+            val validOperatingPointsInPage = allAssetsInPage.mapNotNull(::parseAsset)
+            allPoints.addAll(validOperatingPointsInPage)
+        } while (allAssetsInPage.size == 100)
```

Comment:
Kaikki Ratkon liikennepaikat eivät toteuta ilmoitettua rakennetta, joten sadasta luetusta assetista saattaa vain 99 parsiutua liikennepaikoiksi, joten sivutuksessa ei tutkita onnistuneesti parsittujen liikennepaikkojen lukumäärää, vaan luettujen assettien, näin saadaan kaikki potentiaalisest luettua Ratkosta.

Sellainen miinus tähän liittyy, että epävalidi liikennepaikka ainoastaan ignoorataan.